lib/jaeger/client/injectors.rb

```diff
@@ -16,7 +16,7 @@ class JaegerTextMapCodec
         def self.inject(span_context, carrier)
           carrier['uber-trace-id'] = Injectors.context_as_jaeger_string(span_context)
           span_context.baggage.each do |key, value|
-            carrier["uberctx-#{key}"] = value
+            carrier["uberctx-#{key}"] = CGI.escape(value)
```

Comment:
I think only JaegerRackCodec must be changed. It doesn't seem that java client is encoding the value for non-http protocols: https://github.com/jaegertracing/jaeger-client-java/blob/master/jaeger-core/src/main/java/io/jaegertracing/internal/propagation/TextMapCodec.java#L201-L203

Reply:
mhmmmm, then what I believe is, Java client does not comply with the specification, it says;

> Value: url-encoded string

Reply:
There's a similar check in python: https://github.com/jaegertracing/jaeger-client-python/blob/master/jaeger_client/codecs.py#L68 .
and in node: https://github.com/jaegertracing/jaeger-client-node/blob/master/src/propagators/text_map_codec.js#L37-L39 where urlEncoding is set to false for regular textmap https://github.com/jaegertracing/jaeger-client-node/blob/7a9c1008568b310ec069e72f3d31134391d3138f/src/tracer.js#L89-L95

It looks that url encoding is only needed if it is sent over http

Reply:
mhmmm, maybe a bug in specification then? or just a thing which has to be clarified in documentation. Would you like to ask someone from organisation or just change it for Rack codec for sake of keeping it compliant with other libs?

Reply:
Lets keep it compliant with other libs.

Probably something that should be added to crossdock tests in future though.

Reply:
I've update the pr to only encode/decode values for RackCodec